python/pyiceberg/table/metadata.py

```diff
@@ -69,6 +73,12 @@ def check_partition_specs(values: Dict[str, Any]) -> Dict[str, Any]:
         if spec.spec_id == default_spec_id:
             return values
 
+    # When the table is unpartitioned, we just add the default spec
+    # This makes it optional when defining a table
+    if default_spec_id == UNPARTITIONED_PARTITION_SPEC_ID:
```

Comment:
Unlike sort orders, we didn't reserve an ID for the unpartitioned spec. So unfortunately we can't use logic like this and add the spec. We also don't want to modify the table here.

Reply:
That's a pity, let me remove this